core/windows/test/TimeZoneRulesCompleteTest.kt

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2019-2024 JetBrains s.r.o. and contributors.
+ * Use of this source code is governed by the Apache 2.0 License that can be found in the LICENSE.txt file.
+ */
+
+@file:OptIn(ExperimentalForeignApi::class)
+package kotlinx.datetime.test
+
+import kotlinx.cinterop.*
+import kotlinx.datetime.*
+import kotlinx.datetime.internal.*
+import platform.windows.*
+import kotlin.test.*
+import kotlin.time.Duration.Companion.milliseconds
+
+class TimeZoneRulesCompleteTest {
+
+    /** Tests that all transitions that our system recognizes are actually there. */
+    @Test
+    fun iterateOverAllTimezones() {
```

Comment:
I've tried to run tests locally and `iterateOverAllTimezones` has failed with
```
kotlin.AssertionError: in zone Iran Standard Time, at 2026-03-18T20:30:00Z (our guess at the offset is +04:30). Expected <2026-03-19T00:00>, actual <2026-03-19T01:00>
```

Reply:
My guess is that your Windows isn't up-to-date, because currently, Iran doesn't seem to have DST transitions: both https://en.wikipedia.org/wiki/Iran_Standard_Time and the registry state in my Windows 10 VM think so.

This is probably the same issue as the one with `Morocco Standard Time` and `West Bank Standard Time`: there are some special cases in timezone handling that are not explained with just the registry contents. Could you add `Iran Standard Time` to the list of timezones to skip and see how it goes?

Reply:
> Windows isn't up-to-date

Probably, but where is the discrepancy coming from? Doesn't this test compare the same information obtained with two different ways?

> Could you add Iran Standard Time to the list of timezones to skip and see how it goes?

```
in zone Syria Standard Time, at 2026-10-22T21:00:00Z (our guess at the offset is +02:00). Expected <2026-10-23T00:00>, actual <2026-10-22T23:00>.
```
If I add that one too, everything else passes successfully.

Reply:
My guess is that Windows special-cases some time zones.

* Iran (https://en.wikipedia.org/wiki/Iran_Standard_Time) used to decide on its DST using the Solar Hijri calendar.
* Morocco (https://en.wikipedia.org/wiki/Daylight_saving_time_in_Morocco) used to decide on its DST using the Lunar Hijri calendar.
* The West Bank settlements (https://en.wikipedia.org/wiki/Time_in_the_State_of_Palestine): "IDT will begin on the Friday before the last Sunday of March".
* I don't know which rules were in Syria (https://en.wikipedia.org/wiki/Time_in_Syria) before they abolished the DST, but "sometimes" makes me think the rules were not consistent.

None of this is expressible using the Windows registry, which is why the "pre-existing" transitions are precalculated into the future.

Reply:
ok, let's exclude them from the test